aten/src/ATen/cpu/vec/vec256/vec256_int.h

```diff
@@ -496,34 +496,34 @@ class Vectorized<int16_t> : public Vectorizedi {
   Vectorized<int16_t> le(const Vectorized<int16_t>& other) const;
 };
 
-template <>
-class Vectorized<int8_t> : public Vectorizedi {
+template <typename T>
+class Vectorized<T, std::enable_if_t<std::is_same<T, int8_t>::value || std::is_same<T, uint8_t>::value>> : public Vectorizedi {
```

Comment:
My concern with this style is that you someone down the road might write a function that doesn't work for both types without realizing. On the other hand with a CRTP base class the worst I would expect is that some code gets duplicated between the classes when it could have been shared, so not a correctness issue.

e.g. this is the structure I have in mind
```cpp
// Code shared between uint8_t and int8_t
template <typename T>
class Vectorizedi8: public Vectorizedi {
    static Vectorized<T> blendv(const Vectorized<T>& a, const Vectorized<T>& b,
                                 const Vectorized<T>& mask) {
      return _mm256_blendv_epi8(a.values, b.values, mask.values);
    }
};

// Code that is specific to one type
template<>
class Vectorized<uint8_t>: public Vectorizedi8<uint8_t> {
  Vectorized<uint8_t> abs() const {
     return *this;
  }
};

template<>
class Vectorized<int8_t>: public Vectorizedi8<int8_t> {
  Vectorized<int8_t> abs() const {
     return _mm256_abs_epi8(values);
  }
};